hom/utils.py

```diff
@@ -48,63 +53,73 @@ def get_channel_by_id(
     )
 
 
-def get_channel_by_name(bot, discord_guild_id: int, channel_name: str, channel_type: str = "text"):
-    if discord_guild_id not in [guild.id for guild in bot.guilds]:
+def get_channel_by_name(
+    bot: commands.Bot | discord.Client,
+    guild_id: int,
+    channel_name: str,
+    channel_type: str = "text",
+) -> t.Optional[discord.guild.GuildChannel]:
+    if not any(guild_id == guild.id for guild in bot.guilds):
+        return None
+
+    if not (guild := bot.get_guild(guild_id)):
```

Comment:
Does this assign guild and then check that guild is "something" and not "nothing"? If so, very cool. If not, the "guild.channels" below is lacking context.

Reply:
Yes that's exactly what it does.